### ISSUE TITLE
DC and low-frequency numerical guards

### ISSUE THREAD

**Issue by gvcallen** [paramrf/paramrf#63]
Part of #59

`area: numerics, tests` · `size: S` · `depends: #62, #68`

Split out of the original combined ticket: the scikit-rf validation suite that
used to live here is now its own issue, so the guards can land without waiting on
a full cross-model sweep.

### Context

Two singularities become reachable once sweeps include DC:

- **Skin terms.** `sqrt(w)` has value 0 at ω = 0 but derivative -> ∞. Any sweep
  touching DC yields NaN gradients even though the forward pass looks fine.
- **Conduction term.** `sigma/(w*eps_0)` from #60 is singular at ω = 0.

The remedy throughout is the double-`where` pattern, so value and gradient both
stay finite:

```python
safe_w = jnp.where(w > 0, w, 1.0)
result = jnp.where(w > 0, f(safe_w), limit_value)
```

The inner `where` is what keeps the gradient clean — a single `where` around a
singular expression still produces NaN under reverse-mode AD.

### What is already guarded

#60 and #61 applied the pattern in the materials package and in the immittance
result. Do not redo these; they are the reference implementations to match:

- `AbstractDielectric._add_conduction` — the `sigma/(w*eps_0)` term.
- `BulkConductor.skin_depth` and `BulkConductor.surface_impedance`.
- `ImmittanceResult._per_w`, which recovers L and C as `Im(Z)/w` and carries the
  lowest non-zero frequency into the DC entry.

`AbstractUniformLine.y` separately floors `gL` at a machine-epsilon threshold,
which is the older precedent for this style of guard.

### What is left

Unguarded `sqrt` and division by frequency in the line models:

- `DatasheetLine.immittance` — `sqrt_w = jnp.sqrt(w)` feeding `alpha_c`.
- `PhysicalLine.immittance` — `jnp.sqrt(f / self.f_A)` feeding `A_dB`.
- Whatever #62's dispersion stage and #68's conductor restructure introduce; both
  land before this, so audit their output rather than the code quoted above.
- Any remaining dielectric or conductor implementation added after #60.

### Acceptance criteria

- A `Frequency` starting at 0 Hz gives finite `s` for every line model.
- `jax.grad` of a scalar loss over that sweep is finite for every line model and
  every material model.
- The DC check is a parametrised test over all line classes, so future models
  inherit it. Same for the material classes.
- Where a guard already exists, the test asserts it rather than adding a second
  one.

**Comment by gvcallen** [comment]
Notes from #61 (PR #66), plus two findings that change the scope here.

## 4a is half done, and the audit list needs repointing

The double-`where` guards named in the audit have mostly moved. `TescheCoaxialFormulation`'s `L_skin`/`R_skin` and `WheelerMicrostripFormulation`'s R are gone as separate expressions — both now come from `BulkConductor.surface_impedance`, which is already guarded, as are `skin_depth` and `AbstractDielectric.with_conduction`. `ImmittanceResult`'s derived L and C are guarded too.

Still unguarded, confirmed by test: `DatasheetLine` (`jnp.sqrt(w)`) and `PhysicalLine` (`jnp.sqrt(f / self.fA)`).

## But guarding those is not sufficient, and 4a's framing understates the problem

A sweep starting at 0 Hz currently gives non-finite `s` and NaN gradients for **every** physical line, including coax, whose Z and Y are both perfectly finite at DC:

| model | `s` finite at DC | `grad` finite |
|---|---|---|
| `RLGCLine` (G = 1e-4) | yes | yes |
| `CoaxialLine` | no | no |
| `MicrostripLine` | no | no |
| `PhysicalLine` | no | no |
| `DatasheetLine` | no | no |

The cause is not a `sqrt(w)`. It is `zc = sqrt(Z/Y)` in `AbstractImmittanceLine.zc_and_gammaL`: at ω = 0 a line with no static conductance has `Y = 0` and `Z = 0` exactly, so Zc is `sqrt(0/0)`. `RLGCLine` only survives because its G is non-zero.

So this issue's remark that "Issue 2c removes the L and C division by ω, so this is narrower than it would otherwise have been" is not right. 2c did remove that division, and the immittance itself is now clean at DC — the singularity simply moved into the Zc/γ map, where it needs its own limit. Zc → ∞ as ω → 0 for a lossless line; γ → 0. Deciding what `s` should be there is a modelling call worth making explicitly rather than by threshold.

## A real bug this surfaced: the static conductance vanishes at DC

`AbstractDielectric.with_conduction` zeroes the `sigma/(w*eps_0)` term at ω = 0, which is right for εr itself (it diverges), but the shunt admittance is `Y = jω·ε₀·εr`, and that product has a **finite** DC limit of σ. Zeroing the term inside εr throws it away, so a dielectric with a static conductivity reports `Y(0) = 0` when it should report `2πσ/ln(b/a)` for coax:

```python
line = CoaxialLine(dielectric=ConstantDielectric(epr=2.25, tand=1e-3, sigma=1e-6), ...)
line.immittance(Frequency(start=0.0, ...)).Y[0]   # 0j, should be 2*pi*sigma/ln(b/a)
```

That is the one physically-motivated escape from the `Y = 0` singularity above, and it is currently unavailable. The immittance form is exactly where it should be fixed — `Y = jω·ε₀·εr' + σ` has no singularity at all — but it means the fix belongs in the material/line seam, not only in a guard.

## 4b: scikit-rf is guidance, not ground truth

Flagging before this suite gets built, because the framing "a single external ground truth" invites the wrong reflex when a comparison disagrees. scikit-rf makes its own approximations and modelling choices, and has been wrong before. A disagreement must be explained before any tolerance moves — read the scikit-rf source rather than its docstring, and check both sides against the cited paper.

This is not hypothetical; the two skrf tests landed in #66 already hit it, and in the direction that flatters scikit-rf:

- **`TescheCoaxialFormulation` implements only part of Tesche.** scikit-rf's `model='tesche'` is his eq. (14), `Z = Rdc + Zhf/(1 + Zhf/(jw*Lint))`, an equivalent circuit carrying the DC resistance and the internal inductance of a solid rod (`Lint = mu/(8*pi)`). ParamRF implements only the `Zhf = Zs/(2*pi*r)` term — the high-frequency asymptote. The two agree to ~0.7% at 1 GHz for RG-58 and diverge without bound below the skin-depth transition, where ours tends to R = 0 rather than Rdc. Our class is named for Tesche and cites Tesche (2007) while implementing his asymptote, which is exactly the attribution drift the References convention exists to catch. Directly relevant to 4a: Rdc is what makes the DC limit physical.
- The opposite case also exists: **`MLine(model='wheeler')` is Wheeler's own closed form, where ParamRF uses the Hammerstad simplification of it.** ~0.5% on Zc, ~1.4% on β. Neither side is wrong; they are different published approximations, and the test says so.

Suggest 4b's acceptance criterion be strengthened from "any case where ParamRF deliberately differs from scikit-rf is documented in the test with the reason" to requiring that *every* tolerance above float noise names which implementation is approximating and why. The rule is now recorded in `CLAUDE.md`.

Either the Tesche gap should be closed here (implement eq. (14), which needs `Lint` and `Rdc` and therefore conductor geometry, so it interacts with the `surface_impedance`-plus-radius seam from #60) or it should get its own issue. It is not a tolerance question.
